django/thunderstore/community/views/community.py

```diff
@@ -0,0 +1,85 @@
+from django.conf import settings
+from django.db.models import BigIntegerField, Count, Sum
+from django.shortcuts import redirect
+from django.views import View
+from django.views.generic.list import ListView
+
+from thunderstore.cache.cache import CacheBustCondition
+from thunderstore.cache.pagination import CachedPaginator
+from thunderstore.community.models.community_site import CommunitySite
+from thunderstore.repository.models.package import Package
+
+# Should be divisible by 4 and 3
+MODS_PER_PAGE = 24
+
+
+class CommunitiesView(ListView):
+    model = CommunitySite
+    paginate_by = MODS_PER_PAGE
+    paginator_class = CachedPaginator
+    template_name = "community/communities_list.html"
+
+    def get_base_queryset(self):
+        return self.model.objects.exclude(is_listed=False)
+
+    def get_page_title(self):
+        return "Communities"
+
+    def get_cache_vary(self):
+        return "communities"
+
+    def get_full_cache_vary(self):
+        cache_vary = self.get_cache_vary()
+        return cache_vary
+
+    def order_queryset(self, queryset):
+        return (
+            queryset.prefetch_related()
+            .annotate(
+                total_downloads=Sum(
+                    "community__package_listings__package__versions__downloads",
+                    output_field=BigIntegerField(),
+                ),
+                package_count=Count("community__package_listings"),
```

Comment:
Not sure if it's relevant here, since this is only used for sorting and not actually shown to users, but I guess this would also include inactive packages/listings?

Reply:
This `package_count` isn't actually used anywhere, so I'm just gonna remove it. And use it the one in `get_context_data` anyway all of the queries are going to be cached on some level so 🤷